src/Renderer/ThreeExtended/FeaturesUtils.js

```diff
@@ -0,0 +1,146 @@
+            squaredEpsilon = n;
+        }
+    }
+    return closestPoint;
+}
+
+function pointIsInsidePolygon(point, polygonPoints) {
+    // ray-casting algorithm based on
+    // http://www.ecse.rpi.edu/Homepages/wrf/Research/Short_Notes/pnpoly.html
+
+    const x = point._values[0];
+    const y = point._values[1];
+
+    let inside = false;
+    // in first j is last point of polygon
+    // for each segment of the polygon (j is i -1)
+    for (let i = 0, j = polygonPoints.length - 1; i < polygonPoints.length; j = i++) {
+        const xi = polygonPoints[i]._values[0];
+        const yi = polygonPoints[i]._values[1];
+        const xj = polygonPoints[j]._values[0];
+        const yj = polygonPoints[j]._values[1];
+
+        // isIntersect semi-infinite ray horizontally with polygon's edge
+        const isIntersect = ((yi > y) != (yj > y))
+            && (x < (xj - xi) * (y - yi) / (yj - yi) + xi);
+        if (isIntersect) {
+            inside = !inside;
+        }
+    }
+
+    return inside;
+}
+
+
+function getFeatureAtCoordinate(coordinate, type, coordinates, epsilon, properties, result) {
+    if (type == 'linestring' && pointIsOverLine(coordinate, coordinates, epsilon)) {
+        result.lines.push({ properties, coordinates });
+    } else if (type == 'polygon' && pointIsInsidePolygon(coordinate, coordinates)) {
+        result.polygons.push({ properties, coordinates });
+    } else if (type == 'point') {
+        const closestPoint = getClosestPointsIndice(coordinate, coordinates, epsilon);
+        if (closestPoint) {
+            result.points.push({ properties, coordinates: closestPoint });
+        }
+    }
+}
+export default {
+    /**
+     * filters the features that are under the coordinate
+     *
+     * @param      {Coordinates}  coordinate  the coordinate for the filter condition
+     * @param      {Features}  features  features to filter
+     * @param      {number}  epsilon  tolerance around the coordinate (in coordinate's unit)
+     * @return     {array}  array of filters features
+     */
+    getFeaturesAtCoordinate(coordinate, features, epsilon = 0.1) {
+        const result = { points: [], lines: [], polygons: [] };
+        if (features.geometries) {
+            if (features.extent && !features.extent.isPointInside(coordinate, epsilon)) {
+                return result;
+            }
+            for (const feature of features.geometries) {
+                if (feature.extent && !feature.extent.isPointInside(coordinate, epsilon)) {
+                    continue;
+                }
+                /* eslint-disable guard-for-in */
+                for (const id in feature.featureVertices) {
+                    const polygon = feature.featureVertices[id];
+                    if (polygon.extent && !polygon.extent.isPointInside(coordinate, epsilon)) {
+                        continue;
+                    }
+                    const properties = features.features[id].properties.properties;
```

Comment:
Where does this double `properties.properties` come from?

Reply:
I don't sure understand your question
It come from your PR on the features

Reply:
The first `properties` comes from the PR yes (https://github.com/iTowns/itowns/blob/master/src/Renderer/ThreeExtended/GeoJSON2Feature.js#L164): it's where all custom properties are copied from the geojson.

But I'm afraid that the second comes from your data - like this https://github.com/iTowns/itowns/pull/470/files#diff-8966359153adaa7d05280e380a2a2dbbR13

Does this PR still work if you rename 'properties' in your example geojson to 'orange' for instance?

Reply:
`properties` is in [geojson's specification](https://gist.github.com/virtualandy/1233401#file-geojson-spec-1-1-L231)

Reply:
Oohh I didn't know that :)

In this case, the parser should be fixed, no need to create this properties.properties oddity.
It'll probably wait another PR though....

Reply:
But since we have 2 PR in progress (this one + #519) using this attribute it would be nice to fix this before merging them.

Reply:
I'll add a PR

Reply:
see #527